### PR TITLE
[B-04409] New Sharded Get

### PR DESCRIPTION
**Pull request by neonphog**
DEPENDS ON https://github.com/holochain/holochain/pull/870
SEE https://github.com/holochain/holochain/pull/884 for mock unit test

### CODE REVIEW

File: crates/holochain_p2p/src/spawn/actor.rs [hunk 1]
```diff
@@ -1087,22 +1083,20 @@ impl HolochainP2pHandler for HolochainP2pActor {
...
-                    from_agent,
-                    basis,
-                    remote_agent_count: Some(1),
-                    timeout_ms: options.timeout_ms,
-                    as_race: false,
-                    race_timeout_ms: options.timeout_ms,
-                    payload,
-                })
-                .await?;
+            input.max_remote_agent_count = 1;
```
Comment (freesig) [inline crates/holochain_p2p/src/spawn/actor.rs:1098]: What's the reason for not attempting more then one?

File: crates/kitsune_p2p/types/src/config.rs [hunk 2]
```diff
@@ -100,11 +100,14 @@ pub mod tuning_params_struct {
         /// Default timeout for rpc single. [Default: 30s]
         default_rpc_single_timeout_ms: u32 = 1000 * 30,
 
-        /// Default agent count for rpc multi. [Default: 2]
-        default_rpc_multi_remote_agent_count: u32 = 2,
-
-        /// Default timeout for rpc multi. [Default: 30s]
-        default_rpc_multi_timeout_ms: u32 = 1000 * 30,
+        /// Default agent count for rpc multi. [Default: 3]
+        default_rpc_multi_remote_agent_count: u8 = 3,
```
Comment (freesig) [inline crates/kitsune_p2p/types/src/config.rs:104]: Is this what is overwritten to 1? Maybe that's a little misleading?

**Review by freesig — APPROVED** [review]
Looks good to me. Just some minor suggestions

File: crates/holochain_p2p/src/spawn/actor.rs (hunk 1 above)
Comment (neonphog) [reply crates/holochain_p2p/src/spawn/actor.rs:1098]: I actually have no idea... We set the remote count to 1 in the old code, so I was just trying not to change anything.

File: crates/kitsune_p2p/types/src/config.rs (hunk 2 above)
Comment (neonphog) [reply crates/kitsune_p2p/types/src/config.rs:104]: It's only overwritten for `get_links` and `get_agent_activity`..

`get` and `get_meta` still use the default of 3.

**Pull request merged**